libc/test/include/generic-math-macros_test.cpp

```diff
@@ -0,0 +1,87 @@
+//===-- Unittests for generic-math-macros ---------------------------------===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDSList-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#include "test/UnitTest/Test.h"
+
+/*
+ * The intent of this test is validate that the generic math macros operate as
+ * intended
+ */
+#include "include/llvm-libc-macros/generic-math-macros.h"
+
+// INF can be defined as a number with zeroed out mantissa and 1s in the 
+// exponent
+static uint32_t pos_inf_bits = 0x7F800000;
```

Comment:
You should create a generic test class to test for different floating point types, similar to: https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/smoke/FAbsTest.h

All the special typed constants will be automatically available: https://github.com/llvm/llvm-project/blob/main/libc/test/UnitTest/FPMatcher.h#L69

Reply:
This sounds good. Since we're testing 4 macros, should that mean 4 classes for each macro and a `_test.cpp` file for each floating point type similar to what is done for `fabs` here:
- [`fabs_test.cpp`](https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/smoke/fabs_test.cpp)
- [`fabsf128_test.cpp`](https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/smoke/fabsf128_test.cpp)
- [`fabsf16_test.cpp`](https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/smoke/fabsf16_test.cpp)
- [`fabsf_test.cpp`](https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/smoke/fabsf_test.cpp)
- [`fabsl_test.cpp`](https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/smoke/fabsl_test.cpp)

Reply:
Also curious on the proper way to use the `LibcFPTestHelpers` lib that has `FAbsTest.h`. Looks like in the [`libc/test/src/math/CMakeLists.txt`](https://github.com/llvm/llvm-project/blob/main/libc/test/src/math/CMakeLists.txt) file the `add_fp_unittest` CMake function is used and I also saw some files just linking against `LibcFPTestHelpers`. So far using the latter is working in this case